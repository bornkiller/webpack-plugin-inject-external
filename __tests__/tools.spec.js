/**
 * @description - tools unit tests
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

// packages
const path = require('path');
// internal
const tools = require('../lib/tools');

describe('tools suits', () => {
  it('should parse standard externals definition', () => {
    const target = path.resolve(
      __dirname,
      '__fixture__',
      'bootcdn.externals.yml'
    );

    expect(
      tools.parseExternalDefinition(target, 'development')
    ).toMatchSnapshot();

    expect(
      tools.parseExternalDefinition(target, 'production')
    ).toMatchSnapshot();
  });

  it('should create standard HTML AST', () => {
    const modules = ['react', 'moment'];
    const dictionary = {
      react: [
        {
          url:
            'https://lib.baomitu.com/react-dom/16.10.1/umd/react.development.js',
        },
      ],
      moment: [
        {
          url: 'https://cdn.bootcss.com/moment.js/2.22.1/moment.min.js',
          integrity:
            'sha384-fYxN7HsDOBRo1wT/NSZ0LkoNlcXvpDpFy6WzB42LxuKAX7sBwgo7vuins+E1HCaw',
        },
      ],
    };
    const result = tools.createTree(modules, dictionary);

    expect(result).toMatchSnapshot();
  });
});
