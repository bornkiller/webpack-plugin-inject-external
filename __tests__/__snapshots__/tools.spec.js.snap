// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tools suits should create standard HTML AST 1`] = `
Object {
  "body": Array [
    Object {
      "attributes": Object {
        "crossorigin": "anonymous",
        "src": "https://lib.baomitu.com/react-dom/16.10.1/umd/react.development.js",
        "type": "text/javascript",
      },
      "closeTag": true,
      "tagName": "script",
    },
    Object {
      "attributes": Object {
        "crossorigin": "anonymous",
        "integrity": "sha384-fYxN7HsDOBRo1wT/NSZ0LkoNlcXvpDpFy6WzB42LxuKAX7sBwgo7vuins+E1HCaw",
        "src": "https://cdn.bootcss.com/moment.js/2.22.1/moment.min.js",
        "type": "text/javascript",
      },
      "closeTag": true,
      "tagName": "script",
    },
  ],
  "head": Array [],
}
`;

exports[`tools suits should parse standard externals definition 1`] = `
Object {
  "moment": Array [
    Object {
      "url": "https://cdn.bootcss.com/moment.js/2.22.1/moment.js",
    },
    Object {
      "url": "https://cdn.bootcss.com/moment.js/2.22.1/locale/zh-cn.js",
    },
  ],
  "react": Array [
    Object {
      "url": "https://lib.baomitu.com/react-dom/16.10.1/umd/react-dom.development.js",
    },
  ],
  "web-animations-js": Array [
    Object {
      "url": "https://unpkg.com/web-animations-js@2.3.1/web-animations.min.js",
    },
  ],
}
`;

exports[`tools suits should parse standard externals definition 2`] = `
Object {
  "moment": Array [
    Object {
      "integrity": "sha384-fYxN7HsDOBRo1wT/NSZ0LkoNlcXvpDpFy6WzB42LxuKAX7sBwgo7vuins+E1HCaw",
      "url": "https://cdn.bootcss.com/moment.js/2.22.1/moment.min.js",
    },
    Object {
      "integrity": "sha384-XjUTsP+pYBX4Kwg40gPvhWcRGKZh9gUpDForgg5bwsnyNU+VabKhRX6XkyM6fLk4",
      "url": "https://cdn.bootcss.com/moment.js/2.22.1/locale/zh-cn.js",
    },
  ],
  "react": Array [
    Object {
      "integrity": "sha384-s+4sCJEFimwQu2safB6vclCnbG1xrLnO3YaCIFfqQah5bdnJm9w2su1pT0TQltwW",
      "url": "https://cdn.bootcss.com/react/16.3.2/umd/react.production.min.js",
    },
  ],
  "web-animations-js": Array [
    Object {
      "url": "https://unpkg.com/web-animations-js@2.3.1/web-animations.min.js",
    },
  ],
}
`;
